client/app/src/network/NetworkService.js

```diff
@@ -176,32 +181,42 @@
 
     function getFromPeer(api) {
       var deferred = $q.defer();
-      peer.lastConnection = new Date();
-      $http({
-        url: peer.ip + api,
-        method: 'GET',
-        headers: {
-          'Content-Type': 'application/json',
-          'os': 'ark-desktop',
-          'version': clientVersion,
-          'port': 1,
-          'nethash': network.nethash
-        },
-        timeout: 5000
-      }).then(
-        function(resp) {
-          deferred.resolve(resp.data);
-          peer.isConnected = true;
-          peer.delay = new Date().getTime() - peer.lastConnection.getTime();
-          connection.notify(peer);
-        },
-        function(resp) {
-          deferred.reject("Peer disconnected");
-          peer.isConnected = false;
-          peer.error = resp.statusText || "Peer Timeout after 5s";
-          connection.notify(peer);
+      timeService.getTimestamp().then(
```

Comment:
this will request the time server on each peer call, so it would be better to get timestamp directly from peer response because the network loop pick up the transactions list every 10 seconds

Reply:
Actually looking at this more closely, I don't believe this change is required at all. The peer.delay is simply the latency to make the peer connection. Shouldn't need any server timestamp and local timestamp works fine.